### PR TITLE
CORTX-27592 add a test script for m0ctgdump

### PR DESCRIPTION
**Pull request by huanghua78**
The test script can be run in 2 modes:
(1) Test mode: ./ctgdump_test.sh test
    Motr services are started with parity group setting (N == K).
    An index is created and key/values pairs are inserted.
    All CTG stores of the index are dumped and compared.
(2) Compare mode:
    ./ctgdump_test.sh comp be_stob_path cas_ep proc_fid index_fid
    This mode only dumps and compares all CTG stores of the spcified index.

    example: ./ctgdump_tesh.sh comp /var/motr/systest-766208/ libfab:172.16.179.134@tcp:12345:33:900 0x7200000000000001:0 7800000000012345:12345

Signed-off-by: Sining Wu <sining.wu@seagate.com>
Signed-off-by: Hua Huang <hua.huang@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by huanghua78** [comment]
This work is mostly done by @siningwuseagate .

File: cas/m0ctgdump-howto.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+    ```
+	
+5.  Run the m0ctgdump
+
+    Here is the script to run m-ctgdump and its output. **m0ctgdump** takes the same arguments
+    as m0mkfs and m0d. These arguments can be retrieved from the output of `hctl status`.
+    The most important arguments are:
+    ```Text
+    -d: In a Motr cluster, use the default value: /etc/motr/disks-ios.conf
+    -D: Use the prefex "/var/motr/" + process_id + "/db"
```
Comment (ivan-alekhin) [inline cas/m0ctgdump-howto.md:252]: s/prefex/prefix/g

**Review by ivan-alekhin — APPROVED** [review]
+1.
What I did: looked through the patch, followed the document to run m0ctgdump, ran the system test.
What I did not do: did not check if the patch follows the coding style, did not check correctness of the markdown document formatting.
What would be good to improve later on: to add the system test to the list of xperior tests so that it will be automatically run on the CI.

The patch looks good to me from the perspective of the needs of the DTM team.
@huanghua78, @siningwuseagate , thank you for your help!

**Review by siningwuseagate — APPROVED** [review]

**Comment by huanghua78** [comment]
> What would be good to improve later on: to add the system test to the list of xperior tests so that it will be automatically run on the CI.

This will be added in another PR.

File: cas/m0ctgdump-howto.md (hunk 1 above)
Comment (huanghua78) [reply cas/m0ctgdump-howto.md:252]: OK.

**Pull request merged**